### PR TITLE
Add Release watcher Chart in insight agent

### PR DESCRIPTION
**Pull request by makoscafee**
This PR aims at adding release watcher in insight agents. Few things to figure out before is merged

- [x]  What are the permission needed to run this job
- [x]  Question on the output format file, As the syntax used will it create output in a file?.

### CODE REVIEW

File: stable/insights-agent/templates/release-watcher/_job.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - -u
+    - https://charts.fairwinds.com/stable
+    - -u
+    - https://charts.fairwinds.com/incubator
+    - -u
+    - https://kubernetes-charts.storage.googleapis.com
+    - -u
+    - https://kubernetes-charts-incubator.storage.googleapis.com
+    - -o
+    - /output/release-watcher.json
```
Comment (mjhuber) [inline stable/insights-agent/templates/release-watcher/_job.yaml:24]: `file` should be passed to the `-o` flag.  To specify a path, you can use the `--file-path` flag.  Ex:
```
- -o
- file
- --file-path
- /output/release-watcher.json
```

**Review by mjhuber — COMMENTED** [review]
nice work @makoscafee !  I added a small comment inline.  Also, to answer your questions in the PR:

* What are the permission needed to run this job
The `edit` ClusterRole has the required permissions to run properly.  Will that be too permissive to use in your use case?

* Question on the output format file, As the syntax used will it create output in a file?
I added a small comment to update the flags used to output to a file.  The json format in the file will be of type []ReleaseOutput:
```
type ReleaseOutput struct {
	ReleaseName   string `json:"release"`
	Version       string `json:"version"`
	NewestVersion string `json:"newest"`
	IsOld         bool   `json:"outdated"`
}
```

**Comment by makoscafee** [comment]
Thanks, @mjhuber, I will update the chart as per suggestion.
@rbren what do you think about this? 👇 
>   The `edit` ClusterRole has the required permissions to run properly.  Will that be too permissive to use in your use case?

File: stable/insights-agent/templates/release-watcher/job.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{{- if .Values.releasewatcher.enabled -}} 
+{{- $_ := set . "Label" "releasewatcher" }}
```
Comment (rbren) [inline stable/insights-agent/templates/release-watcher/job.yaml]: This should be `release-watcher`, to be consistent with cronjob.yaml

**Review by rbren — COMMENTED** [review]
Sorry I'm late to the party. @mjhuber `edit` does seem excessive, since we're just reading data. Is there a particular reason RW needs `edit` rather than something more pared down?

@makoscafee let's go with `edit` for now - I think we should stay consistent with the RBAC used in the `release-watcher` project.

File: stable/insights-agent/templates/release-watcher/_job.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {{- toYaml .Values.releasewatcher.resources | nindent 4 }}
+  securityContext:
+    readOnlyRootFilesystem: true
+    allowPrivilegeEscalation: false
+    runAsNonRoot: true
+    runAsUser: 1200
+    capabilities:
+      drop:
+        - ALL
+{{- $_ := set . "Label" "relase-watcher" }}
```
Comment (rbren) [inline stable/insights-agent/templates/release-watcher/_job.yaml]: `s/relase/release/`

File: stable/insights-agent/templates/release-watcher/_job.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{{- define "releasewatcher.job.spec" -}}
+restartPolicy: Never
+serviceAccountName: {{ include "insights-agent.fullname" . }}
+volumes:
+- name: output
+  emptyDir: {}
+containers:
+- name: relase-watcher-audit
```
Comment (rbren) [inline stable/insights-agent/templates/release-watcher/_job.yaml]: `s/relase/release/`

Let's just call this `release-watcher`. The `-audit` suffixes we've been adding aren't really that helpful.

**Review by rbren — COMMENTED** [review]
I'm having an issue running this in my KIND cluster.

From the RW logs, I'm seeing:
```
$ k logs release-watcher-lpcdm  -c relase-watcher-audit
I1206 20:30:48.772547       1 cluster.go:27] Got 0 installed releases in the cluster
```
But `helm ls` gives
```
NAME            REVISION        UPDATED                         STATUS          CHART                   APP VERSION      NAMESPACE
cert-manager    1               Fri Dec  6 16:30:46 2019        DEPLOYED        cert-manager-v0.11.0    v0.11.0          cert-manager
insights-agent  1               Fri Dec  6 20:28:29 2019        DEPLOYED        insights-agent-0.9.6    0.4.0            insights-agent
metrics-server  1               Fri Dec  6 16:30:52 2019        DEPLOYED        metrics-server-2.8.8    0.3.5            metrics-server
nginx-ingress   1               Fri Dec  6 16:30:42 2019        DEPLOYED        nginx-ingress-1.25.0    0.26.1           nginx-ingress
```

@mjhuber any ideas there?

**Review by rbren — COMMENTED** [review]
Chart doesn't seem to work as-is, but I think it's due to the `relase` typo

**Comment by rbren** [comment]
@makoscafee I'm getting `500 error: Couldn't parse report: Unknown report type: release-watcher` when I run this report against the staging cluster

**Comment by makoscafee** [comment]
Ops let me look into it.. this is happening in the `release-watcher` job or in the `insight` side?

**Comment by makoscafee** [comment]
ok I got the fix for this https://github.com/FairwindsOps/Insights/pull/334

**Comment by rbren** [comment]
Hmm I'm getting a 400 error running this against master. Working on getting access to the logs to get more info

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
Gonna merge this so I can start doing some QA

**Pull request merged**